Quaver.Shared/Screens/Select/SelectScreen.cs

```diff
@@ -601,15 +601,15 @@ private void DeleteSelected()
                 {
                     selectedMapset.Maps.ForEach(MapDatabaseCache.RemoveMap);
                     AvailableMapsets.RemoveAt(selectedMapsetIndex);
-                    MapManager.Mapsets.Remove(selectedMapset);
+                    selectedMapset.Maps.ForEach(x => MapManager.Mapsets.RemoveAll(y => y.Maps.Exists(z => x.Md5Checksum == z.Md5Checksum)));
                     view.MapsetScrollContainer.InitializeWithNewSets();
                     view.MapsetScrollContainer.SelectMapset(Math.Min(selectedMapsetIndex, AvailableMapsets.Count - 1));
                 }
                 else
                 {
                     MapDatabaseCache.RemoveMap(selectedDifficulty);
                     selectedMapset.Maps.RemoveAt(selectedDifficultyIndex);
-                    MapManager.Mapsets[selectedMapsetIndex].Maps.Remove(selectedDifficulty);
+                    MapManager.Mapsets.RemoveAll(x => x.Maps.Exists(y => y.Md5Checksum == selectedDifficulty.Md5Checksum));
```

Comment:
yeah great let's delete the entire mapset instead of just 1 dif

Reply:
1. find mapset in MapManager.Mapsets with Directory == selectedMapset.Directory
2. in that particular mapset remove difficulty with md5 == selectedDifficulty.md5